src/script/script.cpp

```diff
@@ -305,28 +327,50 @@ std::string CScript::ToString() const
 
 CScript::ScriptType CScript::GetType() const
 {
-    if (this->IsPayToPublicKeyHash())
-        return CScript::P2PKH;
-    if (this->IsPayToScriptHash())
-        return CScript::P2SH;
+    if (IsPayToPublicKeyHash())
+        return CScript::ScriptType::P2PKH;
+    if (IsPayToScriptHash())
+        return CScript::ScriptType::P2SH;
+    if (IsPayToPublicKey())
+        return CScript::ScriptType::P2PK;
     // We don't know this script type
-    return CScript::UNKNOWN;
+    return CScript::ScriptType::UNKNOWN;
 }
 
 uint160 CScript::AddressHash() const
 {
     // where the address bytes begin depends on the script type
-    int start;
-    if (this->IsPayToPublicKeyHash())
+    int start, size;
+    bool isPayToPublicKey = false;
+    if (IsPayToPublicKeyHash())
+    {
         start = 3;
-    else if (this->IsPayToScriptHash())
+        size = 20;
+    }
+    else if (IsPayToScriptHash())
+    {
         start = 2;
+        size = 20;
+    }
+    else if (IsPayToPublicKey())
+    {
+        start = 1;
+        size = 33;
+        isPayToPublicKey = true;
+    }
     else {
         // unknown script type; return zeros (this can happen)
         vector<unsigned char> hashBytes;
         hashBytes.resize(20, 0);
         return uint160(hashBytes);
     }
-    vector<unsigned char> hashBytes(this->begin() + start, this->begin() + start + 20);
+    
+    vector<unsigned char> hashBytes(this->begin() + start, this->begin() + start + size);
```

Comment:
I understand it's not really in the scope of this PR, but I see an inconsistency in this PR when using the `std` namespace. Given that it should be real quick, I'd take this chance to fix it.

My personal preference, and also the most common style in our codebase, goes for being explicit with `std::` each time it's necessary, and removing the `using namespace std` at the beginning of this file.

Reply:
Done (actually the `using namespace std` was just redundant and useless)